vnext/Microsoft.ReactNative/Utils/BatchingEventEmitter.h

```diff
@@ -13,10 +13,11 @@
 
 namespace winrt::Microsoft::ReactNative::implementation {
 struct BatchedEvent {
+  winrt::hstring eventEmitterName;
   winrt::hstring emitterMethod;
-  int64_t tag{};
   winrt::hstring eventName;
-  JSValue eventObject;
+  int64_t coalescingKey;
+  folly::dynamic params;
```

Comment:
>folly::dynamic params; [](http://example.com/codeflow?start=2&length=22)

Why to switch to dynamic from JSValue? The JSValueTreeWriter and JSValueTreeReader can be used as  IJSValueWriter and IJSValueReader for JSValue.

Reply:
I talked to Eric a little bit about this earlier, and recommend current scheme of taking only stack duration ownership of the delegate (const ref). With the copy happening at dynamic serialization, before going to thread boundary. The sink is dynamic, so dynamic internal to the boundary avoids intermediate JSValue.

Originally the code took long-term ownership of the delegate, but I was concerned we had existing user-code providing delegates which capture by reference, which would be unsafe to store past stack duration. I think it also might have assumed delegates are free-threaded.

Reply:
Yup. Once we change to JSValueWriter from folly deeper in the stack, we can refactor this.